replay2/memory_remapper/memory_remapper.cc

```diff
@@ -14,4 +14,107 @@
 
 #include "memory_remapper.h"
 
-void foo(int var) { return; }
+namespace agi {
+namespace replay2 {
+
+	void MarkDeadAddressRange(const ReplayAddressRange& replayAddressRange) {
+		const std::byte dead[2] = { std::byte(0xDE), std::byte(0xAD) };
+		for(int i = 0; i < replayAddressRange.length(); ++i) {
+			replayAddressRange.baseAddress().bytePtr()[i] = dead[i %2];
+		}
+	}
+
+	ReplayAddress MemoryRemapper::AddMapping(const MemoryObservation& observation) {
+
+		const CaptureAddress& captureAddress = observation.captureAddress();
+
+		auto replayAddressRangeAndOffset = findReplayAddressRangeAndOffset(captureAddress);
+		if(replayAddressRangeAndOffset.first != nullptr) {
+			throw AddressAlreadyMappedException();
+		}
+
+		const size_t mappingLength = observation.resourceGenerator()->length();
```

Comment:
Similar concern with memory_remapper_tests.cc:77, looking at the code this should actually be called 'resourceLength' rather than 'mappingLength', I think the non-obvious dependency here makes things error prone.

Reply:
I can rename this if you like.